modules/@apostrophecms/ui/ui/apos/components/AposTreeRows.vue

```diff
@@ -162,13 +164,39 @@ export default {
       return true;
     }
   },
+  mounted() {
+    // Use $nextTick to make sure attributes like `clientHeight` are settled.
+    this.$nextTick(() => {
+      const branches = this.$el.querySelectorAll('[data-apos-tree-branch]');
+
+      branches.forEach(branch => {
+        // Add padding to the max-height to avoid needing a `resize`
+        // event listener updating values.
+        const height = branch.clientHeight + 20;
+        branch.setAttribute('data-apos-tree-branch', `${height}px`);
+        branch.style.maxHeight = `${height}px`;
+      });
+    });
```

Comment:
Could this be simplified using a `ref`? https://vuejs.org/v2/guide/components-edge-cases.html#Accessing-Child-Component-Instances-amp-Child-Elements

Might save you from having to store the attribute in two places.

Reply:
+1 for refs when this kind of wizardry can't be avoided.

Reply:
I'm not sure how that would work in this case since they are generated.